crates/sc-consensus-subspace-rpc/Cargo.toml

```diff
@@ -16,6 +16,7 @@ targets = ["x86_64-unknown-linux-gnu"]
 async-oneshot = "0.5.9"
 futures = "0.3.31"
 futures-timer = "3.0.3"
+itertools = "0.11.0"
```

Comment:
This isn't a new dependency, because it's already a transitive dependency via sp_core. Also happy to just do an extra allocation instead.

Reply:
I don't have strong opinion, though I was always reluctant to use `itertools` for some reason, it felt like the library provides very convenient tools that can't in fact be implemented efficiently, hence it stimulated me to think harder about data structures and simplify things in a way that doesn't require itertools. I can't say this is completely rational, but that was the case a few times when I was trying to do something inefficient and ultimately found a different more efficient way.

Reply:
I just had a look at the code, and `Itertools::chunks()` is complex and unnecessarily generic internally. It also uses a `RefCell`, which disables some optimisations, but we don’t use that functionality.

But `StreamExt::ready_chunks()` is a pretty simple loop, and Rust is good at optimising iterations and streams.

So I think I’ll replace `stream::iter(…chunks())` with `stream::iter(…).ready_chunks()` to remove the itertools dependency, and maybe reduce the buffer size to avoid allocating that memory when it’s not needed. (Understandably, `StreamExt::ready_chunks()` pre-allocates enough space for the whole buffer.)